### PR TITLE
skiffify the demo

### PR DESCRIPTION
**Pull request by joelgrus**

### CODE REVIEW

File: .skiff/cloudbuild-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    'rollout', 'status',
+    '-f', 'deployment.json'
+  ]
+  env: [
+    'CLOUDSDK_COMPUTE_ZONE=us-west1',
+    'CLOUDSDK_CONTAINER_CLUSTER=skiff-prod'
+  ]
+  dir: '.skiff'
+substitutions:
+  _ENV: staging
```
Comment (schmmd) [inline .skiff/cloudbuild-deploy.yaml]: I assume this was copied and gets overridden if we specify `prod` in Google Cloud Build.

File: .skiff/webapp.jsonnet [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+//  - The service definition uses them to find the pods it directs traffic to.
+local labels = {
+    app: config.appName,
+    env: env,
+    contact: config.contact
+};
+
+local model_labels(model_name) = labels + {
+    model: model_name,
+    role: 'model-server'
```
Comment (schmmd) [inline .skiff/webapp.jsonnet]: @codeviking what is this `role` for?  We'd like to add a comment describing it.

File: .skiff/webapp.jsonnet [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+                        name: config.appName,
+                        image: image,
+                        args: [ '--no-models' ],
+                        readinessProbe: healthCheck,
+                        livenessProbe: healthCheck,
+                        resources: {
+                            requests: {
+                                // Our machines currently have 2 vCPUs, so this
+                                // will allow 4 apps to run per machine
+                                cpu: '0.5',
```
Comment (schmmd) [inline .skiff/webapp.jsonnet]: There's a small amount of overhead for Kubernetes services--so this won't actually allow 4 apps per machine but rather 3.

Comment (schmmd) [reply .skiff/webapp.jsonnet]: Maybe change this to `0.4`?

Comment (schmmd) [reply .skiff/webapp.jsonnet]: Is this for the root application?  Maybe this can just be less...

File: app.py [hunk 4]
```diff
@@ -293,18 +301,32 @@ def info() -> Response:  # pylint: disable=unused-variable
...
+        logger.info(f"setting up default routes for {model_name}")
         app.add_url_rule(f"/{model_name}", view_func=return_page)
         app.add_url_rule(f"/{model_name}/<permalink>", view_func=return_page)
 
+
+    @app.route('/', defaults={ 'path': '' })
     @app.route('/<path:path>')
     def static_proxy(path: str) -> Response: # pylint: disable=unused-variable
-        return send_from_directory(build_dir, path)
+        if os.path.isfile(os.path.join(build_dir, path)):
```
Comment (schmmd) [inline app.py]: Would you comment what the `else` case means?  I don't understand why this change is necessary.

File: .skiff/cloudbuild-deploy.yaml (hunk 1 above)
Comment (codeviking) [reply .skiff/cloudbuild-deploy.yaml]: Yup!

File: .skiff/webapp.jsonnet (hunk 2 above)
Comment (codeviking) [reply .skiff/webapp.jsonnet]: The `role` describe the type of job each part of the system does, so that the relevant Kubernetes service can select things of that type.

Would this comment as described be helpful?:

File: .skiff/webapp.jsonnet (hunk 3 above)
Comment (codeviking) [reply .skiff/webapp.jsonnet]: The AllenNLP demo has quite a few different applications. I'd recommend this is be set on a case by case basis for y'alls configuration, since it might be different than the standard use case.

That said for the template, we should likely scale it down. I cut https://github.com/allenai/skiff-template/issues/86 to follow up.

File: app.py (hunk 4 above)
Comment (codeviking) [reply app.py]: There's a comment below, does that not clarify:

```
            # Send the index.html page back to the client as a catch-all, since
            # we're an SPA and JavaScript acts to handle routes the server
            # doesn't.
```

So, the gist is:

1. Serve the file on disk if one exists
2. If the file on disk doesn't exist, return `index.html`, which references JavaScript that then attempts to handle the route

Let me know if that's a better explanation and I can add it.
